packages/hardhat-core/src/types/config.ts

```diff
@@ -25,6 +26,18 @@ export type NetworkUserConfig =
   | HardhatNetworkUserConfig
   | HttpNetworkUserConfig;
 
+export interface HardforkHistoryUserConfig {
+  [hardforkName: string]: number /* block number */;
+}
+
+export interface HardhatNetworkChainUserConfig {
+  hardforkHistory?: HardforkHistoryUserConfig;
+}
+
+export interface HardhatNetworkChainUserConfigByChainId {
```

Comment:
I'd try to respect the `UserConfig` suffix. What about `HardhatNetworkChainByChainIdUserConfig`?

Reply:
`HardhatNetworkChainsUserConfig`

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/f59333aebda4a5a77fd995da2ca7b60125b2b7bf